test/src/dash-adapter.spec.js

```diff
@@ -231,6 +254,97 @@ describe('DashAdapter: load', () => {
   });
 });
 
+describe('DashAdapter: targetBuffer', () => {
+  let video, dashInstance, config;
+
+  beforeEach(() => {
+    video = document.createElement('video');
+    config = {playback: {options: {html5: {dash: {}}}}};
+  });
+
+  afterEach(done => {
+    dashInstance.destroy().then(() => {
+      dashInstance = null;
+      done();
+    });
+  });
+  it('should check targetBuffer in VOD far from end of stream', done => {
+    try {
+      dashInstance = DashAdapter.createAdapter(video, vodSource, config);
+      video.addEventListener(EventType.PLAYING, () => {
+        dashInstance.targetBuffer.should.equal(
+          dashInstance._shaka.getConfiguration().streaming.bufferingGoal + dashInstance._getCurrentSegmentLength()
+        );
+        done();
+      });
+      dashInstance.load().then(() => {
+        video.play();
+      });
+    } catch (e) {
+      done(e);
+    }
+  });
+
+  it('should check targetBuffer in VOD close to end of stream (time left to end of stream is less than targetBuffer)', done => {
+    try {
+      dashInstance = DashAdapter.createAdapter(video, vodSource, config);
+      video.addEventListener(EventType.PLAYING, () => {
+        video.currentTime = video.duration - 1;
+        video.addEventListener(EventType.SEEKED, () => {
+          dashInstance.targetBuffer.should.equal(video.duration - video.currentTime);
+          done();
+        });
+      });
+      dashInstance.load().then(() => {
+        video.play();
+      });
+    } catch (e) {
+      done(e);
+    }
+  });
+
+  it('should check targetBuffer in LIVE', done => {
+    try {
+      dashInstance = DashAdapter.createAdapter(video, liveSource, config);
+      video.addEventListener(EventType.PLAYING, () => {
+        dashInstance._shaka.configure({streaming: {bufferingGoal: 120}});
```

Comment:
Is there a reason to configure on `playing`? we don't do that on the player so I think it'll be better to do that before.

Reply:
done